### PR TITLE
Added metrics list for NooBaa under OCS

### PR DESCRIPTION
**Pull request by jeniawhite**

### CODE REVIEW

**Comment by jeniawhite** [comment]
@squat @s-urbaniak 
Any updates? 
Thank you.

File: jsonnet/telemeter/metrics.jsonnet [hunk 1]
```diff
@@ -94,4 +94,14 @@
   '{__name__="job:ceph_pools_iops_bytes:total"}',
   // job:ceph_versions_running:count is the total count of ceph cluster versions running.
   '{__name__="job:ceph_versions_running:count"}',
-]
+  // job:NooBaa_buckets_health:sum is the number of unhealthy buckets
+  '{__name__="job:NooBaa_buckets_health:sum"}',
+  // job:NooBaa_bucket_count:sum is the number buckets
```
Comment (anmolsachan) [inline jsonnet/telemeter/metrics.jsonnet]: Can this be rephrased ? Dosen't make complete sense

File: jsonnet/telemeter/metrics.jsonnet [hunk 2]
```diff
@@ -94,4 +94,14 @@
...
+  '{__name__="job:NooBaa_bucket_count:sum"}',
+  // job:NooBaa_total_object_count:sum is the number of objects
+  '{__name__="job:NooBaa_total_object_count:sum"}',
+  // NooBaa_accounts_num gives the count of NooBaa's accounts.
+  '{__name__="NooBaa_accounts_num"}',
+  // NooBaa_total_usage gives the total usage of NooBaa's storage in bytes.
+  '{__name__="NooBaa_total_usage"}',
+]
```
Comment (anmolsachan) [inline jsonnet/telemeter/metrics.jsonnet]: Add line

File: jsonnet/telemeter/metrics.jsonnet (hunk 1 above)
Comment (jeniawhite) [reply jsonnet/telemeter/metrics.jsonnet]: Can you suggest a more suitable name?

File: jsonnet/telemeter/metrics.jsonnet [hunk 3]
```diff
@@ -94,4 +94,14 @@
   '{__name__="job:ceph_pools_iops_bytes:total"}',
   // job:ceph_versions_running:count is the total count of ceph cluster versions running.
   '{__name__="job:ceph_versions_running:count"}',
+  // job:NooBaa_buckets_health:sum is the number of unhealthy buckets
+  '{__name__="job:NooBaa_total_unhealthy_buckets:sum"}',
+  // job:NooBaa_bucket_count:sum is the number buckets
```
Comment (anmolsachan) [inline jsonnet/telemeter/metrics.jsonnet]: job:NooBaa_bucket_count:sum is the **total** number **of Noobaa** buckets

File: jsonnet/telemeter/metrics.jsonnet [hunk 4]
```diff
@@ -94,4 +94,14 @@
   '{__name__="job:ceph_pools_iops_bytes:total"}',
   // job:ceph_versions_running:count is the total count of ceph cluster versions running.
   '{__name__="job:ceph_versions_running:count"}',
+  // job:NooBaa_buckets_health:sum is the number of unhealthy buckets
```
Comment (anmolsachan) [inline jsonnet/telemeter/metrics.jsonnet]: job:NooBaa_buckets_health:sum is the number of unhealthy **Noobaa** buckets

File: jsonnet/telemeter/metrics.jsonnet [hunk 5]
```diff
@@ -94,4 +94,14 @@
...
+  '{__name__="job:NooBaa_total_unhealthy_buckets:sum"}',
+  // job:NooBaa_bucket_count:sum is the number buckets
+  '{__name__="job:NooBaa_bucket_count:sum"}',
+  // job:NooBaa_total_object_count:sum is the number of objects
```
Comment (anmolsachan) [inline jsonnet/telemeter/metrics.jsonnet]: job:NooBaa_total_object_count:sum is the number of **Noobaa** objects

**Comment by anmolsachan** [comment]
@jeniawhite  Thanks for the updates.

Can you please attach Prometheus console screenshots for the metrics to confirm that the cardinality of these metrics is 1.

**Comment by jeniawhite** [comment]
@anmolsachan 
I'm on it. Trying to deploy everything so I would be able to screenshot the metrics.

**Comment by squat** [comment]
once you have a running openshift cluster with all of these new metrics being collected by the monitoring stack, please export the KUBECONFIG env var and run `make test/timeseries.txt -B` to collect sample metrics and commit them to the repo.

File: docs/data-collection.md [hunk 6]
```diff
@@ -102,6 +102,16 @@ For the OpenShift 4 Developer Preview we will be sending back these exact attrib
   '{__name__="job:ceph_pools_iops_bytes:total"}',
   // job:ceph_versions_running:count is the total count of ceph cluster versions running.
   '{__name__="job:ceph_versions_running:count"}',
+  // job:NooBaa_total_unhealthy_buckets:sum is the total number of unhealthy NooBaa buckets
+  '{__name__="job:NooBaa_total_unhealthy_buckets:sum"}',
```
Comment (squat) [inline docs/data-collection.md]: by convention, all metric names should be lowercase including proper nouns.

Comment (squat) [reply docs/data-collection.md]: :)

**Comment by jeniawhite** [comment]
@anmolsachan 
Here are the screenshots.
If you need any additional info please let me know.
Screen Shot 2019-07-30 at 16 05 02
Screen Shot 2019-07-30 at 16 06 11
Screen Shot 2019-07-30 at 16 06 24
Screen Shot 2019-07-30 at 16 06 36

**Comment by jeniawhite** [comment]
@squat 
Updated the metric recordings.

**Comment by anmolsachan** [comment]
@jeniawhite Can you please rebase this PR .

**Comment by smarterclayton** [comment]
Approved for addition to the product (once sur is happy with the mechanics).

**Comment by jeniawhite** [comment]
@squat @smarterclayton 
Everything is rebased and ready.

**Comment by squat** [comment]
@jeniawhite the change set looks fine but this pr introduces ~20 commits with an odd commit history. Could you squash them to just the essential 2?
1 add new metric
2 regenerate

**Comment by jeniawhite** [comment]
@squat 
Squashed the commits. Thanks.

**Pull request merged**